head.go

```diff
@@ -524,6 +595,9 @@ func (h *headIndexReader) Series(ref uint32) (labels.Labels, []*ChunkMeta, error
 			MinTime: c.minTime,
 			MaxTime: c.maxTime,
 			Ref:     (uint64(ref) << 32) | uint64(i),
+
+			deleted: deleted,
+			dranges: dranges,
```

Comment:
I see how this is getting tricky. Maybe it makes sense to not consider tombstones at this level. It's the low-level interface to access data we have. Given that the data is still there, just extended by tombstones, maybe it should remain queryable.

At higher level abstractions we can then combine that info with our known tombstones to provide the end result. We could just extend `IndexReader` by a `Tombstone(ref)` method. For the disk index this now means  it's multiple files. Or, maybe cleaner, make tombstones its own additional concept on the same level as chunks and index.